File: 1805.number-of-different-integers-in-a-string/solution_1_pass.py
    def numDifferentIntegers(self, word: str) -> int:
        res, startIndex = set(), 0
        for i, c in enumerate(word):
            if c.isdigit():
                if startIndex < i and word[startIndex] == '0':
                    startIndex += 1
            else:
                if startIndex < i:
                    res.add(word[startIndex:i])
                startIndex = i + 1

        if startIndex < len(word):
            res.add(word[startIndex:])
        return len(res)

File: 1805.number-of-different-integers-in-a-string/solution_1_pass.cs
public class Solution {
    public int NumDifferentIntegers(string word)
    {
        var startIndex = 0;
        var set = new HashSet<string>();

        for (var i = 0; i < word.Length; i++)
        {
            if (word[i] >= '0' && word[i] <= '9')
            {
                startIndex += startIndex < i && word[startIndex] == '0' ? 1 : 0;
            }
            else
            {
                if (startIndex < i)
                {
                    set.Add(word.Substring(startIndex, i - startIndex));
                }

                startIndex = i + 1;
            }
        }

        if (startIndex < word.Length)
        {
            set.Add(word.Substring(startIndex));
        }

        return set.Count;
    }
}